elrond-wasm-debug/src/execute_mandos.rs

```diff
@@ -373,6 +437,105 @@ fn check_state(accounts: &mandos::CheckAccounts, state: &mut BlockchainMock) {
 					);
 				}
 			}
+
+			match &expected_account.esdt {
+				Some(CheckEsdt::Equal(eq)) => {
```

Comment:
There is a Checkable trait, that is used precisely for this. Implementing Checkable for CheckEsdt gives you the "check" method, where you can move the checking logic, with the match and all.
The point is to have less code here, where there is already plenty, and have at least the checking done somewhere else.